casper/src/main/resources/Either.rho

```diff
@@ -108,10 +108,15 @@ new  Either, rs(`rho:registry:insertSigned:secp256k1`), uriOut, rl(`rho:registry
         ((false, _), _)               => return!(a)
         (_, (false, _))               => return!(b)
         ((true, va), (true, vb)) => {
-          new ret in {
-            f!(va, vb, *ret) |
-            for (@v <- ret) {
-              return!((true, v))
+          new ret, stdout(`rho:io:stdout`) in {
+            match *f {
+              { for (x, y, r <- fName) { _ } } => {
+                *f |
+                fName!(va, vb, *ret) |
+                for (@v <- ret) {
+                  return!((true, v))
+                }
+              }
```

Comment:
Doesn't this change the behavior of map2 whenever you're not passing in a `for`?

Reply:
yes, it does. There's only one usage though.